### ISSUE TITLE
Path::components strip `.` under the hood

### ISSUE THREAD

**Issue by hyf0** [hyf0/sugar_path#16]
This behavior is part of [spec](https://github.com/rust-lang/rfcs/blob/master/text/0474-path-reform.md#summary).  

https://github.com/rust-lang/rust/pull/23229

This makes the following code 

https://github.com/hyf0/sugar_path/blob/41fce88b6161694b1b533020a49e675ccb0f77d8/src/utils.rs#L29-L31

unreliable for detecting whether the dots are resolved.

**Comment by hyf0** [comment]
I'm going to keep this issue open. The issue block sugar_path to implementing on-demand normalization.

https://github.com/hyf0/sugar_path/blob/41fce88b6161694b1b533020a49e675ccb0f77d8/src/sugar_path.rs#L18

I saw how path-absolutize handles this, and I am not going to use it.

https://github.com/magiclen/path-absolutize/blob/c2966357fced68b60852333a6c0ca01250925b22/src/unix.rs#L63-L66

https://github.com/magiclen/path-absolutize/blob/c2966357fced68b60852333a6c0ca01250925b22/src/unix.rs#L86-L88

To summarize, I would remove the on-demand normalization feature. But the return type would still be `Cow` in case of reducing future changes.
